lib/active_model/serializer/adapter.rb

```diff
@@ -94,6 +95,7 @@ def include_meta(json)
         json[meta_key] = meta if meta
         json
       end
+      ##### AbstractAdapter
```

Comment:
There were circular dependencies related to the Adapter class having subclasses in that used Adapter as the namespace, something you wouldn't find with a module-based namespace.  While trying to figure that out, I realized that all the methods below here are essentially an `AbstractAdapter`, or what in Rails-world might also be called `Adapter::Base`, and that the methods above it were the only ones that weren't part of the 'AbstractAdapter', and though I ended up removing that refactoring due to the increased complexity of the PR, I left this as a note... I can remove it, but now you know :)